date.go

```diff
@@ -129,13 +137,20 @@ func (d *dateObject) isSet() bool {
 
 func (d *dateObject) unset() {
 	d.msec = timeUnset
+	d.val.__wrapped = nil
```

Comment:
I **think** we'll need to do the same in the `setTimeMs` function within the `if` statement because if we don't, we'll still be using the `__wrapped` time after changing the time.